### PR TITLE
fix: Use HTTP BadRequest instead of InternalServerError

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

Replace this behavior:
image

With this:
image

## References

* #782

### CODE REVIEW

File: source/dotnet/Services/WebApi/V2/BatchController.cs [hunk 1]
```diff
@@ -59,18 +55,15 @@ public class BatchController : ControllerBase
...
         if (new ZonedDateTime(periodEnd, _dateTimeZone).TimeOfDay != LocalTime.Midnight)
-            return StatusCode((int)HttpStatusCode.BadRequest, $"The period end '{periodEnd.ToString()}' must be 1 ms before midnight.");
-
-        if (!_batchRequestDtoValidator.IsValid(batchRequestDto, out var errorMessages))
-            return StatusCode((int)HttpStatusCode.BadRequest, string.Join(" ", errorMessages));
+            throw new BusinessValidationException($"The period end '{periodEnd.ToString()}' must be 1 ms before midnight.");
 
         var batchId = await _batchApplicationService.CreateAsync(batchRequestDto).ConfigureAwait(false);
-        return Ok(batchId);
+        return batchId;
```
Comment (lasrinnil) [inline source/dotnet/Services/WebApi/V2/BatchController.cs:66]: why this change here?

**Review by lasrinnil — APPROVED** [review]

File: source/dotnet/Services/WebApi/V2/BatchController.cs (hunk 1 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/WebApi/V2/BatchController.cs:66]: Because (1) it's simpler, and (2) it allows for a more specific return type (method signature).

File: source/dotnet/Services/WebApi/V2/BatchController.cs [hunk 2]
```diff
@@ -59,18 +55,15 @@ public class BatchController : ControllerBase
     /// <returns>200 Ok with The batch id, or a 400 with an errormessage</returns>
```
Comment (JesperGomes) [inline source/dotnet/Services/WebApi/V2/BatchController.cs:55]: Is this comment stil valid?

Comment (BjarkeMeier) [reply source/dotnet/Services/WebApi/V2/BatchController.cs:55]: Why would it not be?

Comment (JesperGomes) [reply source/dotnet/Services/WebApi/V2/BatchController.cs:55]: Well, this method does not return OK anymore...it returns a Guid?

Comment (BjarkeMeier) [reply source/dotnet/Services/WebApi/V2/BatchController.cs:55]: I consider the code doc of actions to be intended for the HTTP usage. This also aligns with our work to actually make this documentation part of our Open API specification.
